### ISSUE TITLE
Customize images for widgets and shares (opengraph)

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2660]
Following up on https://github.com/opencollective/opencollective/issues/2550
Based on user's feedback from https://opencollective.freshdesk.com/a/tickets/2417 and https://opencollective.freshdesk.com/a/tickets/3111

---

Using collective's logo for widgets and shares on social networks (opengraph) is a good default, but we're a lacking a feature to customize them for users who cares about that.

It is important because we've enforced stricter rules in the new collective page - logo must be squared and displayed on a white background.

# Proposed solution

### Frontend

- [ ] Add a section in `/edit/images` to change the images for the widgets and social networks
- [ ] If a custom opengraph image has been defined, use it to generate `og:image`

### API

I'd argue that we should not blotter the (already big) `Collectives` table, so I'm advocating for the creation of a new `Images` table. We may eventually want to migrate the existing `image`/`backgroundImage` columns in the future, but that's out of scope for this issue.

- [ ] Add a new `Images` table
  - CollectiveId
  - type - ENUM(`OPENGRAPH_BANNER`|`WIDGETS_LOGO`)
  - url
- [ ] Add the corresponding fields in graphql
- [ ] Add the endpoints to upload images

## Alternative solution (or MVP?)

A much simpler alternative would be to just use `settings` with:
- `settings.images.opengraphBanner`
- `settings.images.widgetsLogo`

Which doesn't require any change to the API.

---

Please wait for @znarf's feedback before starting any implementation.
Issue's complexity will be adapted based on that.

**Comment by jdaviescoates** [comment]
I'm here because @Betree thought this might also be a nice solution to my own issue #3604 :)

**Comment by Betree** [comment]
I would love to see a solution for that issue as it's preventing many collectives to use their communication assets properly. Unless there's a blocker, I'll try to implement the very simple MVP proposed in the issue so that we can offer a workaround for the collectives that need it.
